### PR TITLE
Update powsybl 

### PR DESCRIPTION
**Pull request by phiedw**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Dependency update

**What is the current behavior?** *(You can also link to an open issue here)*
Farao is based on an old version of powsybl 

**What is the new behavior (if this is a feature change)?**
Farao is now based on powsybl 3.7.0

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java [hunk 1]
```diff
@@ -123,7 +135,7 @@ public void testPtdfResultManipulation() {
     public void testNokSensiResult() {
         // When
         SensitivityAnalysisResult sensitivityAnalysisResult = Mockito.mock(SensitivityAnalysisResult.class);
-        Mockito.when(sensitivityAnalysisResult.isOk()).thenReturn(false);
+        //Mockito.when(sensitivityAnalysisResult.isOk()).thenReturn(false);
```
Comment (pet-mit) [inline sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java:138]: keep or delete

File: sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java [hunk 2]
```diff
@@ -123,7 +135,7 @@ public void testPtdfResultManipulation() {
     public void testNokSensiResult() {
```
Comment (pet-mit) [inline sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java:135]: how come this test still tests "not ok" result ?

**Review by pet-mit — CHANGES_REQUESTED** [review]

File: sensitivity-analysis/src/main/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResult.java [hunk 3]
```diff
@@ -67,17 +61,21 @@ public SystematicSensitivityResult() {
 
     public SystematicSensitivityResult completeData(SensitivityAnalysisResult results, boolean afterCra) {
 
-        if (results == null || !results.isOk()) {
```
[the remark this answers is not in the document]
Comment (phiedw) [reply sensitivity-analysis/src/main/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResult.java:70]: The powsybl API no longer has an isOk method

File: sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java (hunk 2 above)
Comment (phiedw) [reply sensitivity-analysis/src/test/java/com/farao_community/farao/sensitivity_analysis/SystematicSensitivityResultTest.java:135]: renamed the test

Now the SensitivityResult's default status is FAILURE, and if we find at least one NaN value in the powsybl result, we set the status to SUCCESS. This test is to make sure the status is indeed FAILURE when there are no values.

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java [hunk 4]
```diff
@@ -234,14 +234,18 @@ private void updateOptimalLeafWithNextDepthBestLeaf(AbstractNetworkPool networkP
                             .forEach(ra -> ra.apply(networkClone, previousDepthOptimalLeaf.getOptimizedSetPoint(ra)));
                         optimizeNextLeafAndUpdate(naCombination, networkClone, networkPool);
                         networkPool.releaseUsedNetwork(networkClone);
-                        TECHNICAL_LOGS.info("Remaining leaves to evaluate: {}", remainingLeaves.decrementAndGet());
                     } else {
                         topLevelLogger.info("Skipping {} optimization because earlier combination fulfills stop criterion.", naCombination.getConcatenatedId());
                         networkPool.releaseUsedNetwork(networkClone);
                     }
-                } catch (Exception e) {
+                    TECHNICAL_LOGS.info("Remaining leaves to evaluate: {}", remainingLeaves.decrementAndGet());
```
Comment (pet-mit) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java:241]: put this line in finally block

**Review by pet-mit — CHANGES_REQUESTED** [review]

**Review by pet-mit — APPROVED** [review]

**Pull request merged**